### PR TITLE
created campaign index featured page relationship

### PR DESCRIPTION
**Pull request by danielfmiranda**
# Description

Link to sample test page: https://foundation-s-featured-c-bfvssc.herokuapp.com/en/campaigns/
Related PRs/issues: https://mozilla-hub.atlassian.net/browse/TP1-27

This pull request introduces a new `featured_page` section to the CampaignIndexPage model. This enhancement enables users to manually curate the list of pages rendered on the campaign index page at `/campaigns/`.

Additionally, this PR includes updates to tests and factories to support this new feature.

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-397)

## Steps to test:
1. Visit the CMS edit page at [this link](https://foundation-s-featured-c-bfvssc.herokuapp.com/cms/pages/93/edit/) using the credentials: **admin2/admin2**
2. Make a change to the "featured pages" section (for example, reorder the list or delete/add an item)
3. Confirm that your changes appear correctly in the preview panel
4. Publish your changes
5. Visit https://foundation-s-featured-c-bfvssc.herokuapp.com/en/campaigns/
6. The list of pages shown here should match what you set in the CMS
7. Visit https://foundation-s-featured-c-bfvssc.herokuapp.com/fr/campaigns/
8. Note that the page "Trade tough once" is now showing its localized version "Trade tough once(in french)"
9. If the campaign page is working as expected (load more, etc). Testing is complete!

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/generic_card.html [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {% load wagtailcore_tags wagtailimages_tags %}
 {% if page %}
     {% if not url %}
-        {% pageurl page as url %}
+        {% pageurl page.localized as url %}
```
Comment (danielfmiranda) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/generic_card.html:4]: Calling out that we are typically trying to move away from using .localized in templates, however, this template is shared amongst a couple of other pages as well, and the work necessary to get this template to work with all of them without using .localized would exceed the scope of this ticket.

However, we do have issue https://github.com/MozillaFoundation/foundation.mozilla.org/issues/10018 that captures that work!

**Comment by robdivincenzo** [comment]
Hey @danielfmiranda! I'm running into an issue on step 3. Here's a recording of what I see. Not sure that it's the same case for this issue, but I experienced something similar with sort_order=None by default. I had to adjust the factory to fix it.

https://github.com/MozillaFoundation/foundation.mozilla.org/assets/3083419/a04e36e6-62ab-47ed-b2dd-82bf34844476

**Comment by danielfmiranda** [comment]
Hi @robdivincenzo thanks for taking a look and alerting me to the issue you were experiencing. I have updated the factory and it seems to be working now! Can you please take a look at the new review app and let me know if the issue is fixed for you as well? 

Thanks!

**Review by robdivincenzo — APPROVED** [review]
LGTM, thanks Daniel! 🚀🚀

**Comment by data-sync-user** [comment]
➤ Simon Acosta Torres commented:

This PR was merged on April 17th.

**Pull request merged**